tests/testsuite/features.rs

```diff
@@ -480,11 +480,14 @@ fn no_feature_doesnt_build() {
         .run();
     p.process(&p.bin("foo")).with_stdout("").run();
 
-    p.cargo("build --features bar")
+    p.cargo("build --features bar -v")
         .with_stderr(
             "\
 [COMPILING] bar v0.0.1 ([CWD]/bar)
+[RUNNING] `rustc --crate-name bar [..]
+[DIRTY-MSVC] foo v0.0.1 ([CWD]): the list of features changed
```

Comment:
Haven't looked into this. Why this is a MSVC specified output?

Reply:
This test demonstrates that:
https://github.com/rust-lang/cargo/blob/ab18bd40d5bf6d69689cece1c546a928b0766b64/tests/testsuite/freshness.rs#L447-L513

In particular this comment:

```rs
     // MSVC does not include hash in binary filename, so it gets recompiled. 
```

I'm not sure about what hash that is referring to, but it seems to be important. I can only assume that is because of #7400.